libs/dh/market-participant/feature-edit-actor/src/lib/market-participant-edit-actor/metering-point-type/dh-market-participant-actor-metering-point-type.component.ts

```diff
@@ -0,0 +1,69 @@
+/**
+ * @license
+ * Copyright 2020 Energinet DataHub A/S
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License2");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { CommonModule } from '@angular/common';
+import {
+  Component,
+  EventEmitter,
+  Input,
+  NgModule,
+  OnChanges,
+  Output,
+} from '@angular/core';
+import { FormsModule } from '@angular/forms';
+import { MeteringPointTypeChanges } from '@energinet-datahub/dh/market-participant/data-access-api';
+import {
+  ActorDto,
+  MarketParticipantMeteringPointType,
+} from '@energinet-datahub/dh/shared/domain';
+import { TranslocoModule } from '@ngneat/transloco';
+import { MatListModule } from '@angular/material/list';
+
+@Component({
+  selector: 'dh-market-participant-actor-metering-point-type',
+  templateUrl:
+    './dh-market-participant-actor-metering-point-type.component.html',
+})
+export class DhMarketParticipantActorMeteringPointTypeComponent
+  implements OnChanges
+{
+  @Input() actor: ActorDto | undefined;
+  @Output() hasChanges = new EventEmitter<MeteringPointTypeChanges>();
+  changes: MeteringPointTypeChanges = { meteringPointTypes: [] };
+  availableMeteringPointTypes = Object.values(
+    MarketParticipantMeteringPointType
+  );
+
+  ngOnChanges(): void {
+    if (this.actor !== undefined) {
+      this.changes = {
+        meteringPointTypes: this.actor.meteringPointTypes,
+      };
+      this.hasChanges.emit({ ...this.changes });
```

Comment:
issue: even though this part of the code works, state is synced between the component and the store, and users can edit the status, the pattern of emitting from `ngOnChanges` hook is not desired. (The same pattern is used in other parts of the app as well.)

Removing this line completely, here or in other parts of the app, will cause the feature to not work correctly, so we can't do it just yet. Our suggestion is that we to "go back to the writing board" and come up with an alternative pattern for you to use every time you have the need to sync data between local and store state. This is not going to be part of the current PR but something that we'll come with in the near future.